### PR TITLE
Ci pipeline

### PR DESCRIPTION
**Pull request by AxiomSamarth**
**What this PR does / why we need it**:
This PR adds the CI pipeline scripts & configs to the Equinix Metal repo

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
NONE
```

### CODE REVIEW

File: .ci/pipeline_definitions [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            registry: 'gcr-readwrite'
+            image: 'eu.gcr.io/gardener-project/gardener/machine-controller-manager-provider-equinix-metal'
+    steps:
+      check:
+        image: 'golang:1.13.5'
+      build:
+        image: 'golang:1.13.5'
+        output_dir: 'binary'
+      test:
+        image: 'golang:1.13.5'
```
Comment (prashanth26) [inline .ci/pipeline_definitions]:
```suggestion
      check:
        image: 'golang:1.16.2'
      build:
        image: 'golang:1.16.2'
        output_dir: 'binary'
      test:
        image: 'golang:1.16.2'
```
Probably move the images to [recent golang](https://hub.docker.com/layers/golang/library/golang/1.16.2/images/sha256-d24c4fa66898a1ab5b7aa22ad9a3ed61210d7a1f8af56c684d12b750e5ae7965?context=explore) version.

**Review by prashanth26 — COMMENTED** [review]
/lgtm otherwise

**Review by prashanth26 — APPROVED** [review]
/lgtm

**Comment by deitch** [comment]
lgtm

**Comment by prashanth26** [comment]
@AxiomSamarth - Can you please /squash the commits before merging. I see too many commits. Not sure if it's relavant.

**Comment by AxiomSamarth** [comment]
Rebasing suddenly added too many files into the PR. Did I mess something up?

**Comment by deitch** [comment]
It looks like it is based on the original `main`, prior to my PR #2 merging in. Check `main` back out, pull it so you get the latest, and then rebase on that.

**Comment by deitch** [comment]
@AxiomSamarth fixed it up nicely. I think this is ready for review.

**Review by prashanth26 — APPROVED** [review]
/lgtm

**Comment by AxiomSamarth** [comment]
> @AxiomSamarth fixed it up nicely. I think this is ready for review.

Thanks to you for the guidance 😄

**Pull request merged**